tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,332 @@
+
+    Polymer({
+      is: "tf-pr-curve-card",
+      properties: {
+        runs: Array,
+        tag: String,
+        // For each run, the card will display the PR curve at this step or the
+        // one closest to it, but less than it.
+        runToStepCap: Object,
+        _runToPrCurveEntry: {
+          type: Object,
+          value: {},
+          notify: true,
+        },
+        // A list of runs with an available step. Used to populate the table of
+        // steps per run.
+        _runsWithStepAvailable: {
+          type: Array,
+          computed: "_computeRunsWithStepAvailable(runs, _runToPrCurveEntry)",
+        },
+        _runToData: Object,
+        _colorScaleFunction: {
+          type: Object,  // function: string => string
+          value: () => ({scale: runsColorScale}),
+        },
+        requestManager: Object,
+        _canceller: {
+          type: Object,
+          value: () => new Canceller(),
+        },
+        _attached: Boolean,
+        // The value field is a function that returns a function because Polymer
+        // will actually call the value field if the field is a function.
+        // However, we actually want the value itself to be a function.
+        _xComponentsCreationMethod: {
+          type: Object,
+          readOnly: true,
+          value: () => (() => {
+            const scale = new Plottable.Scales.Linear();
+            return {
+              scale: scale,
+              axis: new Plottable.Axes.Numeric(scale, 'bottom'),
+              accessor: d => d.recall,
+            };
+          }),
+        },
+        _yValueAccessor: {
+          type: Object,
+          readOnly: true,
+          // This function returns a function because polymer calls the outer
+          // function to compute the value. We actually want the value of this
+          // property to be the inner function.
+          value: () => (d => d.precision),
+        },
+        loading: Boolean,
+        _tooltipColumns: {
+          type: Array,
+          readOnly: true,
+          value: () => {
+            const valueFormatter = ChartHelpers.multiscaleFormatter(
+                ChartHelpers.Y_TOOLTIP_FORMATTER_PRECISION);
+            const formatValueOrNaN =
+                (x) => isNaN(x) ? 'NaN' : valueFormatter(x);
+            return [
+              {
+                title: 'Name',
+                computeFunction: (d) => d.dataset.metadata().name,
+              },
+              {
+                title: 'Precision',
+                computeFunction: (d) => formatValueOrNaN(d.datum.precision),
+              },
+              {
+                title: 'Recall',
+                computeFunction: (d) => formatValueOrNaN(d.datum.recall),
+              }];
+          },
+        }
+      },
+      observers: [
+          'reload(run, tag)',
+          '_runsChanged(_attached, runs.*)',
+          '_setChartData(_runToPrCurveEntry.*)',
```

Comment:
I'm glad to hear that this worked and that you like the design. Could you please confirm that this works properly when you have a set of runs like `["foo", "foo.bar", "foo.baz", "foo.", "."]` and similar? (Note that `"."` will be an extra common run name.) The call to ``this.set(`_runToPrCurveEntry.${run}`)`` seems like it might not do what you want in those cases—or maybe it does.